### PR TITLE
Add extra threshold information to pre-built object detection curves

### PR DESCRIPTION
**Pull request by gordonhart**
Fixes KOL-2855

### CODE REVIEW

**Review by marko-polo-cheno — DISMISSED** [review]
Nice 👍, lmk if you want me to change unit tests

File: kolena/_experimental/object_detection/utils.py [hunk 1]
```diff
@@ -108,14 +108,22 @@ def _compute_threshold_curve(
...
-        return Curve(x=recalls, y=precisions, label=curve_label) if len(recalls) >= 2 else None
+        return (
+            Curve(x=thresholds, y=f1s, label=curve_label, extra=dict(Precision=precisions, Recall=recalls))
+            if len(f1s) >= 2
+            else None
+        )
+
+    # add a point to start the PR curve on the vertical axis if needed
+    if 0.0 not in recalls:
+        minpos = recalls.index(min(recalls))
```
Comment (marko-polo-cheno) [inline kolena/_experimental/object_detection/utils.py:125]: might need to change f1s and thresholds if there is a length complaint

File: kolena/_experimental/object_detection/utils.py [hunk 2]
```diff
@@ -108,14 +108,22 @@ def _compute_threshold_curve(
...
+        precisions.append(precisions[minpos])
+        recalls.append(0.0)
```
Comment (gordonhart) [inline kolena/_experimental/object_detection/utils.py:127]: Will also need to add `f1s`, `thresholds` points here.

File: kolena/_experimental/object_detection/utils.py (hunk 1 above)
Comment (gordonhart) [reply kolena/_experimental/object_detection/utils.py:125]: Do you mind taking a look at the unit test failures? Feel free to push to this branch directly.

File: kolena/_experimental/object_detection/utils.py [hunk 3]
```diff
@@ -77,15 +77,16 @@ def _compute_threshold_curve(
...
+    potential_thresholds = np.unique(y_score[y_score >= 0.0]).tolist()  # sorts
 
-    if len(thresholds) >= 501:
-        thresholds = list(np.linspace(min(thresholds), max(thresholds), 501))
+    if len(potential_thresholds) >= 501:
+        potential_thresholds = list(np.linspace(min(potential_thresholds), max(potential_thresholds), 501))
 
     precisions: List[float] = []
     recalls: List[float] = []
+    thresholds: List[float] = []
```
Comment (marko-polo-cheno) [inline kolena/_experimental/object_detection/utils.py:87]: It is better that the points of a PR curve also exist on the F1 curve (and vice versa)

**Review by ly-kolena — APPROVED** [review]

**Review by munkyshi — APPROVED** [review]

**Comment by gordonhart** [comment]
Awesome 👍 ([trunk/gordon](https://trunk.kolena.io/gordon/results?modelIds=627&testSuiteId=1333&workflow=Object+Detection&evaluators=N4IgjAzFIFygbAJgOyzPADPAvtoA)) 

Screen Shot 2023-08-25 at 12 09 02 PM

**Pull request merged**